nipype/algorithms/confounds.py

```diff
@@ -1201,20 +1298,56 @@ def compute_noise_components(imgseries, mask_images, num_components,
         # "The covariance matrix C = MMT was constructed and decomposed into its
         # principal components using a singular value decomposition."
         try:
-            u, _, _ = fallback_svd(M, full_matrices=False)
+            u, s, _ = fallback_svd(M, full_matrices=False)
         except np.linalg.LinAlgError:
-            if self.inputs.failure_mode == 'error':
+            if failure_mode == 'error':
                 raise
-            u = np.ones((M.shape[0], num_components), dtype=np.float32) * np.nan
+            if components_criterion >= 1:
+                u = np.full((M.shape[0], components_criterion),
+                            np.nan, dtype=np.float32)
+            else:
+                continue
```

Comment:
If `fallback_svd` fails, `s` will not be defined, which will fail in L1311. If, additionally, `components_criterion < 1`, `u` will not be defined, either. What's the expected behavior in these cases?